carbon-p2-plugin/src/main/java/org/wso2/maven/p2/feature/generate/FeatureGenerator.java

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+
+    /**
+     * Traverse through dependency and dependencyManagement section and populate the project dependencies into
+     * internal bean structures
+     *
+     * @throws IOException
+     * @throws OSGIInformationExtractionException
+     */
+    private void resolveMavenProjectDependencies() throws IOException, OSGIInformationExtractionException {
+        this.log.info("Inspecting maven dependencies.");
+        List<HashMap<String, CarbonArtifact>> artifacts = DependencyResolver.getDependenciesForProject(project,
+                resourceBundle.getRepositorySystem(),
+                resourceBundle.getRemoteRepositories(), resourceBundle.getLocalRepository());
+        dependentBundles = artifacts.get(0);
+        dependentFeatures = artifacts.get(1);
+    }
+
+    /**
+     * Cross check the given project dependencies with the bundles and features specified in the maven plugin
+     * configuration and fill specified bundles and features with extra details taken from project dependencies.
+     *
+     * @throws CarbonArtifactNotFoundException
+     */
+    private void populateRequiredArtifactData() throws CarbonArtifactNotFoundException {
+        populateBundleDataFromCache();
+        populateFeatureDataFromCache();
+    }
+
+    /**
+     * The generated bundle beans using the configuration fed into the plugin contain only fewer details. Thus traverse
+     * through the maven dependencies specified in the pom and update the bundle beans with extra details gathered from
+     * dependencies.
+     *
+     * @throws CarbonArtifactNotFoundException
+     */
+    private void populateBundleDataFromCache() throws CarbonArtifactNotFoundException {
+        for (Bundle bundle : resourceBundle.getBundles()) {
+            String key = bundle.getSymbolicName() + "_" + bundle.getVersion();
+            CarbonArtifact artifact = dependentBundles.get(key);
+            if (artifact == null) {
+                throw new CarbonArtifactNotFoundException("Bundle " + key + " is not found in project dependency list");
+            }
+            artifact.copyTo(bundle);
+        }
+    }
+
+    /**
+     * The generated feature beans using the configuration fed into the plugin contain only fewer details. Thus traverse
+     * through the maven dependencies specified in the pom and update the feature beans with extra details gathered from
+     * dependencies.
+     *
+     * @throws CarbonArtifactNotFoundException
+     */
+    private void populateFeatureDataFromCache() throws CarbonArtifactNotFoundException {
+        for (Feature feature : resourceBundle.getIncludeFeatures()) {
+            String key = feature.getId() + ".feature" + "_" + feature.getVersion();
+            CarbonArtifact artifact = dependentFeatures.get(key);
+            if (artifact == null) {
+                throw new
+                        CarbonArtifactNotFoundException("Feature " + key + " is not found in project dependency list");
```

Comment:
For readability purpose, shall we break this line from string msg than from class name level?